saleor/dashboard/product/views.py

```diff
@@ -711,3 +711,21 @@ def get_variant_label(variant):
         for variant in queryset
     ]
     return JsonResponse({'results': variants})
+
+
+@staff_member_required
```

Comment:
Shouldn't we guard those views with permissions similar to non-ajax views?

Reply:
Good catch. I'm not sure what restrictions are set on the "add voucher" view, but if there is the ability to view the products, then the person having rights to edit vouchers also needs rights to view the products.